// @vuemap/vue-amap/lib/utils/index.js
'use strict';

var guid = require('./guid.js');
var util = require('./util.js');
var eventHelper = require('./eventHelper.js');
var convertHelper = require('./convert-helper.js');
var GPSUtil = require('./GPSUtil.js');
var buildHelper = require('./buildHelper.js');
var makeInstaller = require('./make-installer.js');



exports.guid = guid.default;
exports.convertEventToLowerCase = util.convertEventToLowerCase;
exports.convertLnglat = util.convertLnglat;
exports.eventReg = util.eventReg;
exports.isIndoorMapInstance = util.isIndoorMapInstance;
exports.isLabelsLayerInstance = util.isLabelsLayerInstance;
exports.isMapInstance = util.isMapInstance;
exports.isOverlayGroupInstance = util.isOverlayGroupInstance;
exports.isVectorLayerInstance = util.isVectorLayerInstance;
exports.loadScript = util.loadScript;
exports.upperCamelCase = util.upperCamelCase;
exports.bindInstanceEvent = eventHelper.bindInstanceEvent;
exports.removeInstanceEvent = eventHelper.removeInstanceEvent;
exports.lngLatTo = convertHelper.lngLatTo;
exports.pixelTo = convertHelper.pixelTo;
exports.toBounds = convertHelper.toBounds;
exports.toLngLat = convertHelper.toLngLat;
exports.toPixel = convertHelper.toPixel;
exports.toSize = convertHelper.toSize;
exports.bd09_To_gcj02 = GPSUtil.bd09_To_gcj02;
exports.bd09_To_gps84 = GPSUtil.bd09_To_gps84;
exports.gcj02_To_bd09 = GPSUtil.gcj02_To_bd09;
exports.gcj02_To_gps84 = GPSUtil.gcj02_To_gps84;
exports.gps84_To_bd09 = GPSUtil.gps84_To_bd09;
exports.gps84_To_gcj02 = GPSUtil.gps84_To_gcj02;
exports.lonLatToTileNumbers = GPSUtil.lonLatToTileNumbers;
exports.tileNumbersToLonLat = GPSUtil.tileNumbersToLonLat;
exports.buildProps = buildHelper.buildProps;
exports.commonProps = buildHelper.commonProps;
exports.makeInstaller = makeInstaller.makeInstaller;
